packages/xrpl/test/connection.test.ts

```diff
@@ -0,0 +1,990 @@
+/* eslint-disable max-len -- Some large lines necessary */
+/* eslint-disable max-statements -- test has a lot of statements */
+import net from 'net'
+
+import { assert } from 'chai'
+import {
+  Client,
+  ConnectionError,
+  DisconnectedError,
+  NotConnectedError,
+  ResponseFormatError,
+  XrplError,
+  TimeoutError,
+} from 'xrpl-local'
+import { Connection } from 'xrpl-local/client/connection'
+
+import rippled from './fixtures/rippled'
+import {
+  setupClient,
+  teardownClient,
+  type XrplTestContext,
+} from './setupClient'
+import { assertRejects, ignoreWebSocketDisconnect } from './testUtils'
+
+type GlobalThis = typeof globalThis
+type Global = GlobalThis & {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Necessary for Jest in browser
+  TextEncoder: any
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- Necessary for Jest in browser
+  TextDecoder: any
+}
+declare const global: Global
+
+if (typeof TextDecoder === 'undefined') {
+  // eslint-disable-next-line node/global-require, @typescript-eslint/no-require-imports, node/prefer-global/text-encoder, global-require, @typescript-eslint/no-var-requires -- Needed for Jest
+  global.TextEncoder = require('util').TextEncoder
+  // eslint-disable-next-line node/global-require, @typescript-eslint/no-require-imports, node/prefer-global/text-decoder, global-require, @typescript-eslint/no-var-requires -- Needed for Jest
+  global.TextDecoder = require('util').TextDecoder
+}
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Necessary to get browser info
+const isBrowser = (process as any).browser
+
+let lastSocketKey = 0
+const socketMap: { [socketKey: string]: net.Socket } = {}
+
+async function destroyServer(server: net.Server): Promise<void> {
```

Comment:
There seems to be a similar function with the same name defined in `testUtils`, is there overlap between the two?

Reply:
@JST5000 they're similar but different slightly